bin/memex/hackathon_2016_07/scripts/download_url_for_doc_ids.py

```diff
@@ -0,0 +1,47 @@
+import hashlib
+import os
+import requests
+import sys
+
+# input is like:
+# http://someimageurl docid1 docid2.. docidn
+
+# Download someimageurl to a location based on its sha1 hash
+# Then print out the mapping of the URL to the sha
+
+NUM_RETRIES = 5
+
+if __name__ == '__main__':
+    url, doc_ids = sys.argv[1], sys.argv[1:]
+
+    DATA_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                            '../data')
+
+    i = 0
+    while True:
+        try:
+            r = requests.get(url, stream=True)
+        except Exception:
+            sys.exit(3)
+
+        if (r.status_code >= 400 and r.status_code < 500):
+            sys.exit(1)
+        elif i == NUM_RETRIES:
+            sys.exit(2)
+        elif not r.ok:
+            i += 1
+            continue
+        else:
+            sha1 = hashlib.sha1(r.content).hexdigest()
+
+            try:
+                os.makedirs(os.path.join(DATA_DIR, 'CP1_imageset', sha1[0:3]))
```

Comment:
There is a function in ``smqtk.utils.file_utils`` called ``save_create_dir`` that is for this.